### ISSUE TITLE
Is it possible to have a callback on EFI module loads on a UEFI based guest VM?

### ISSUE THREAD

**Issue by subvert0r** [libvmi/libvmi#1105]
Assume I have a XEN based windows VM that is UEFI based.

My question is, is it possible to somehow have a callback that gets called whenever a UEFI module (Bootmgr.efi, Winload.efi)  gets loaded in the guest VM during boot, using hooks or vmexit monitoring, etc?

**Comment by tklengyel** [comment]
Yes, you can do that. You need to locate the appropriate memory location to breakpoint or place EPT restrictions on to be able to receive your callback but there is nothing special about OVMF in terms of hooking it through LibVMI.

**Comment by tklengyel** [comment]
Sounds like you have to perform some forensic analysis of OVMF to figure out how to interpret it's memory layout, internal control-flow, etc. and figure out where you have to hook, possibly at multiple places and piece the information you need back together. Or depending on the scenario you could also just modify the OVMF that gets loaded to issue you a magic callback without you having to figure out how to inject it.
